gtests/test_nopython_callprofiler.cpp

```diff
@@ -212,5 +212,30 @@ TEST_F(CallProfilerTest, cancel)
     EXPECT_EQ(radix_tree().get_unique_node(), 0);
 }
 
+TEST_F(CallProfilerTest, test_serialization)
+{
+    pProfiler->reset();
+
+    foo1();
+
+    // Example:
+    //  {
+    //      "radix_tree":
+    //      {
+    //          "nodes":[
+    //              {"key":-1,"name":"","data":{"start_time": "0","caller_name": "","total_time": 0,"call_count": 0,"is_running": 0},"children":[0]},
+    //              {"key":0,"name":"void modmesh::detail::foo1()","data":{"start_time": "17745276708555250","caller_name": "void modmesh::detail::foo1()","total_time": 61002916,"call_count": 1,"is_running": 1},"children":[1]},
+    //              {"key":1,"name":"void modmesh::detail::foo2()","data":{"start_time": "17745276708555458","caller_name": "void modmesh::detail::foo2()","total_time": 54002250,"call_count": 1,"is_running": 1},"children":[2]},
+    //              {"key":2,"name":"void modmesh::detail::foo3()","data":{"start_time": "17745276743555833","caller_name": "void modmesh::detail::foo3()","total_time": 19001833,"call_count": 1,"is_running": 1},"children":[]}
+    //          ],
+    //          "current_node":-1,
+    //          "unique_id":3
+    //      }
+    //  }
+
+    std::stringstream ss;
+    pProfiler->serialize(ss);
+}
+
```

Comment:
Here is the simple example of the serialization of the CallProfiler.

Reply:
you only ran the function, but did verify the output.

here is what you can do:
1. have a answer json file
2. run the function and see if the output is equal to the answer.
3. optional, verify the result matching the json schema (see more [here](https://python-jsonschema.readthedocs.io/en/latest/validate/))